web_poet/utils.py

```diff
@@ -1,7 +1,114 @@
+import inspect
 import weakref
 from collections.abc import Iterable
 from functools import wraps
 from typing import Any, List, Optional
+from warnings import warn
+
+
+def _clspath(cls, forced=None):
+    if forced is not None:
+        return forced
+    return f"{cls.__module__}.{cls.__name__}"
+
+
+def _create_deprecated_class(
```

Comment:
Copied from Scrapy, only removing the ability to customize the deprecation warning class, which is here hardcoded as `DeprecationWarning` instead.

Reply:
should we copy tests as well?